lib/matplotlib/backend_tools.py

```diff
@@ -989,12 +989,10 @@ def trigger(self, *args, **kwargs):
                  'help': ToolHelpBase,
                  'copy': ToolCopyToClipboardBase,
                  }
-"""Default tools"""
```

Comment:
Why are these removed? Not sure if they are actually rendered, but those are doc-strings for the attributes.

Reply:
They were removed given @greglucas review comment here: https://github.com/matplotlib/matplotlib/pull/22809#discussion_r846630906

Reply:
@oscargus I'll let you and the other members of the dev team make a call here, but the fact that this does show up in the [the current docs](https://matplotlib.org/devdocs/api/backend_tools_api.html?highlight=default_tools#matplotlib.backend_tools.default_tools) looks unintentional to me. At least I don't see their value here in the docs, but I'm also obviously not an expert in this section of the API.

![not_intentional](https://user-images.githubusercontent.com/5142394/163420046-68b430e7-a6ed-43c7-b16f-28af46b4cf59.png)

These docstring comments were added in 2015 back in PR #3652 by @fariza. So while I sure hope no one ever asks me to explain code that I wrote over 5 years ago :grimacing: maybe they can take a look and give their thoughts.

@greglucas as you suggested removal maybe you can also give your thoughts here.

Reply:
I agree that it is not the most relevant section of the MPL documentation, so I do not see any real issue with letting it go, especially `default_tools`. Just wanted to point out the consequence.

Reply:
Can you please clarify if there is a pre-commit hook that made you turn the doc-string into a comment? (As I assume that the purpose was not to get rid of the documentation.)

Would be good to know in case we at some stage start to enforce this. (Which we probably shouldn't then.)

Reply:
> Just wanted to point out the consequence.

Sure. I totally get that, and I truly appreciate you thinking about this PR so carefully. So thank you for being a great reviewer! :)

Reply:
> Can you please clarify if there is a pre-commit hook that made you turn the doc-string into a comment? (As I assume that the purpose was not to get rid of the documentation.)

That's part of what the [`check-docstring-first` hook](https://github.com/pre-commit/pre-commit-hooks/tree/c7a7ffb9a71a6859cf588f967bb8a208d5871d53#check-docstring-first) does

```
$ git checkout upstream/main -- lib/matplotlib/backend_tools.py
$ pre-commit run check-docstring-first --all-files
check docstring is first.................................................Failed
- hook id: check-docstring-first
- exit code: 1

lib/matplotlib/backend_tools.py:992: Multiple module docstrings (first docstring on line 1).

```

> Would be good to know in case we at some stage start to enforce this. (Which we probably shouldn't then.)

I think the use cases for having a real use for multiple module level docstrings is quite rare. At that point might as well just ignore that particular file for this particular hook.

I'll note again though that I don't have any strong feelings on `check-docstring-first` being used or not used in general &mdash; it is just one of the pre-commit hooks added in PR #21583.

Reply:
That was added just following the general directions that all public variables should be documented. There was no specific intent to it.
And specially since the previous entry shows the same list being added, it is not really needed.

Reply:
Thanks @fariza!

Reply:
I think the question is not whether we should remove these strings (because they do seem to not provide much more information anyway), but whether pre-commit will preclude us from ever documenting constants like this, and whether that is ever going to be something that we need to do.

Reply:
The good news is that whatever you decide the impact is quite low.

* If you decide to keep this hook and then later decide that it is impeding you, then you add the file that is failing to `check-docstring-first`'s specific `exclude`. Or you delete `check-docstring-first`
* If you don't want `check-docstring-first` then we delete it from pre-commit and optionally drop these changes.

So maybe the most relevant question is: can think of any examples where you would want multiple module level docstrings? The hooks all pass so you don't have any else in the codebase. If in the entire project these were the only ones and they were added in 2015 then it seems a safe bet this isn't going to change in the near future unless you already have examples in mind.

Reply:
FYI, check-docstring-first had a long-standing issue for these false positives, see https://github.com/pre-commit/pre-commit-hooks/issues/159. It can be turned off per-file, and does catch the common case where imports sneak above the docstring. Or it can be dropped, either way.

Reply:
I think this affects excess docstrings in examples far more than variable docstrings in the actual library, so I think I'm net positive on keeping it.

Reply:
> so I think I'm net positive on keeping it.

(Sorry, long week and my brain is a bit fuzzy.) Positive on keeping the `check-docstring-first` hook, correct?

Reply:
I'm pretty indifferent on this (as it sounds like most of us are!). I'd lean towards sticking with what you have currently (removing the docstring/comment [and because of that removing the sphinx generated documentation] and leaving the pre-commit check in place).

Reply:
@QuLogic @oscargus @jklymak @greglucas @tacaswell As it seems that there are no strong feelings against keeping the `check-docstring-first` hook and some positive leanings towards it, does anyone have any further questions on this PR?

To keep the noise down on this I've stopped rebasing it against `main` but I have a branch on my fork where I've continued to do so and `pre-commit run --all-files` is still passing with this rebased on `main`'s `HEAD`.